edx_rbac/mixins.py

```diff
@@ -61,3 +65,79 @@ def check_permissions(self, request):
             self.permission_denied(
                 request,
                 message=('MISSING: {}'.format(', '.join(missing_permissions))))
+
+
+class PermissionRequiredForListingMixin(PermissionRequiredMixin):
+    """
+    Mixin for checking permissions against multiple objects.
+    """
+    list_lookup_field = None
+    staff_are_never_forbidden = True
+    allowed_roles = []
+    role_assignment_class = None
+
+    @property
+    def base_queryset(self):
+        raise NotImplementedError
+
+    @cached_property
+    def request_action(self):
+        return getattr(self, 'action', None)
+
+    @cached_property
+    def accessible_identifiers(self):
+        """
+        Cached set of identifiers the requesting user has access to.
+        Returns a set that contains the `ALL_ACCESS_CONTEXT` identifier
+        if the requesting user is a superuser.
+        """
+        identifiers_via_jwt = utils.contexts_accessible_from_request(self.request, self.allowed_roles)
+        identifiers_via_db = set()
+
+        if self.role_assignment_class:
+            identifiers_via_db = utils.contexts_accessible_from_database(
+                self.request.user, self.allowed_roles, self.role_assignment_class
+            )
+
+        if self.request.user.is_superuser:
+            identifiers_via_db.add(utils.ALL_ACCESS_CONTEXT)
+
+        return identifiers_via_jwt | identifiers_via_db
+
+    def check_permissions(self, request):
+        """
+        TODO
+        """
+        if self.request_action == 'list':
+            # Super-users and staff won't get Forbidden responses,
+            # but depending on their assigned roles, staff may
+            # get an empty result set.
+            if request.user.is_superuser:
+                return
+            if request.user.is_staff and self.staff_are_never_forbidden:
+                return
+            if not self.accessible_identifiers:
+                self.permission_denied(request)
+        else:
+            super().check_permissions(request)
+
+    def get_queryset(self):
+        """
+        Expects `self.base_queryset` to be explicitly defined as the "base case"
+        and `self.list_lookup_field` to be defined.
+        You'll most likely want it to be `MyModel.objects.all().order_by('some_field')`.
+        """
+        if not getattr(self, 'base_queryset', None):
+            raise Exception('{} must have a non-null "base_queryset" field.'.format(self.__class__))
+        if not getattr(self, 'list_lookup_field', None):
```

Comment:
is the `list_lookup_field` expected to be a string?

Reply:
Yeah, it's gotta be a string.  I can add an assertion about that somewhere, I think.